services/wireguard/endpoint/create_linux.go

```diff
@@ -40,12 +43,17 @@ func NewConnectionEndpoint(ipResolver ip.Resolver, resourceAllocator *resources.
 	}, nil
 }
 
-// GeneratePrivateKey creates new wireguard private key
-func GeneratePrivateKey() (string, error) {
-	return kernelspace.GeneratePrivateKey()
+func getWGClient() (wgClient wgClient, err error) {
+	if isKernelSpaceSupported() {
+		return kernelspace.NewWireguardClient()
+	}
+
+	log.Info("Wireguard kernel space is not supported. Switching to user space implementation.")
+	return userspace.NewWireguardClient()
 }
 
-// PrivateKeyToPublicKey generates wireguard public key from private key
-func PrivateKeyToPublicKey(key string) (string, error) {
-	return kernelspace.PrivateKeyToPublicKey(key)
+func isKernelSpaceSupported() bool {
+	err := utils.SudoExec("ip", "link", "add", "iswgsupported", "type", "wireguard")
+	_ = utils.SudoExec("ip", "link", "del", "iswgsupported")
+	return err == nil
```

Comment:
Logging of error would be nice - in case we need to figure out why exactly it failed - maybe wg is still supported but some little detail needs to be tweaked.

Reply:
Added debug output.